.github/workflows/template_release_version.yml

```diff
@@ -0,0 +1,41 @@
+name: Release Version Detector
+
+on:
+  workflow_call:
+    outputs:
+      new_version:
+        value: ${{ jobs.new_version.outputs.new_version }}
+
+jobs:
+  new_version:
+
+    name: Get new release version
+    runs-on: ubuntu-22.04
+
+    outputs:
+      new_version: ${{ steps.set_version.outputs.next_version }}
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Detect new release version
+        id: set_version
+        run: |
+          # fetch last tag from git
+          OLD_VERSION=`git describe --tags $(git rev-list --tags --max-count=1)`
+          PARTS=(${OLD_VERSION//./ })
+          DATE=$(date +"%Y%m%d")
+          YEAR=$(date +"%Y")
+          WEEK=$(date +"%-W")
+          PATCH=${PARTS[2]}
```

Comment:
Maybe we should call that "counter"? That's at least what I will refer it to in the ADR. Since it is determined incrementally, it is also more self-explanatory then (vs "patch").